// @salesforce/design-system-react/module/components/data-table/__examples__/advanced-header-row.js
function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function"); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, writable: true, configurable: true } }); if (superClass) _setPrototypeOf(subClass, superClass); }

function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf || function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }

function _createSuper(Derived) { var hasNativeReflectConstruct = _isNativeReflectConstruct(); return function _createSuperInternal() { var Super = _getPrototypeOf(Derived), result; if (hasNativeReflectConstruct) { var NewTarget = _getPrototypeOf(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return _possibleConstructorReturn(this, result); }; }

function _possibleConstructorReturn(self, call) { if (call && (_typeof(call) === "object" || typeof call === "function")) { return call; } return _assertThisInitialized(self); }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function () {})); return true; } catch (e) { return false; } }

function _getPrototypeOf(o) { _getPrototypeOf = Object.setPrototypeOf ? Object.getPrototypeOf : function _getPrototypeOf(o) { return o.__proto__ || Object.getPrototypeOf(o); }; return _getPrototypeOf(o); }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

import React from 'react';
import DataTable from "../../../../components/data-table"; // `~` is replaced with design-system-react at runtime

import DataTableColumn from "../../../../components/data-table/column";
import DataTableCell from "../../../../components/data-table/cell";
import IconSettings from "../../../../components/icon-settings";

var CustomHeaderRow = function CustomHeaderRow(_ref) {
  var columns = _ref.columns,
      item = _ref.item;
  return (
    /*#__PURE__*/
    // +1 colSpan due to selection.
    React.createElement("tr", {
      key: item.id
    }, /*#__PURE__*/React.createElement("th", {
      style: {
        backgroundColor: '#fff',
        padding: '0.5rem 1.25rem'
      },
      id: item.id,
      colSpan: columns.length + 1,
      scope: "colgroup"
    }, /*#__PURE__*/React.createElement("p", {
      style: {
        fontWeight: 700
      },
      role: "heading",
      "aria-level": item.ariaLevel
    }, item.heading)))
  );
};

CustomHeaderRow.displayName = DataTableCell.displayName;

var Example = /*#__PURE__*/function (_React$Component) {
  _inherits(Example, _React$Component);

  var _super = _createSuper(Example);

  function Example() {
    var _this;

    _classCallCheck(this, Example);

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = _super.call.apply(_super, [this].concat(args));

    _defineProperty(_assertThisInitialized(_this), "state", {
      sortColumn: 'opportunityName',
      sortColumnDirection: {
        opportunityName: 'asc'
      },
      items: [{
        type: 'header-row',
        id: 'header-row-example-id-1',
        heading: 'Argentina',
        ariaLevel: 1
      }, {
        id: '8IKSDFZV91',
        headerId: 'header-row-example-id-1',
        opportunityName: 'Mule Systems',
        accountName: 'MuleSoft',
        closeDate: '3/31/17 2:05PM',
        stage: 'Processing',
        confidence: '80%',
        amount: '$14,000',
        contact: 'bob@mulesoft.com'
      }, {
        id: '8IKZHZZV80',
        opportunityName: 'Acme - 1,200 Widgets',
        headerId: 'header-row-example-id-1',
        accountName: 'Acme',
        closeDate: '4/10/15',
        stage: 'Value Proposition',
        confidence: '70%',
        amount: '$25,000,000',
        contact: 'jrogers@acme.com'
      }, {
        id: '5GJOOOPWU7',
        opportunityName: 'Acme - 200 Widgets',
        headerId: 'header-row-example-id-1',
        accountName: 'Acme',
        closeDate: '1/31/15',
        stage: 'Prospecting',
        confidence: '30%',
        amount: '$5,000,000',
        contact: 'bob@acme.com'
      }, {
        type: 'header-row',
        id: 'header-row-example-id-2',
        heading: 'Argentina > Autonomous City of Buenos Aires',
        ariaLevel: 2
      }, {
        id: '8SDFDSZHZZV34',
        headerId: 'header-row-example-id-2',
        opportunityName: 'salesforce.com - 5,000 Widgets',
        accountName: 'MulesSoft',
        closeDate: '10/31/18 4:15AM',
        stage: 'Id. Decision Makers',
        confidence: '35%',
        amount: '$50,000',
        contact: 'chyral@salesforce.com'
      }, {
        type: 'header-row',
        id: 'header-row-example-id-3',
        heading: 'Argentina > Autonomous City of Buenos Aires > Belgrano',
        ariaLevel: 3
      }, {
        id: '8IKZHWERWRV81',
        headerId: 'header-row-example-id-3',
        opportunityName: 'salesforce.com - 1,000 Widgets',
        accountName: 'salesforce.com',
        closeDate: '1/31/15 3:45PM',
        stage: 'Id. Decision Makers',
        confidence: '60%',
        amount: '$25,000',
        contact: 'nathan@salesforce.com'
      }],
      selection: []
    });

    _defineProperty(_assertThisInitialized(_this), "handleChanged", function (event, data) {
      _this.setState({
        selection: data.selection
      });

      console.log(event, data);
    });

    return _this;
  }

  _createClass(Example, [{
    key: "render",
    value: function render() {
      return /*#__PURE__*/React.createElement("div", null, /*#__PURE__*/React.createElement(IconSettings, {
        iconPath: "/assets/icons"
      }, /*#__PURE__*/React.createElement(DataTable, {
        assistiveText: {
          actionsHeader: 'actions',
          columnSort: 'sort this column',
          columnSortedAscending: 'asc',
          columnSortedDescending: 'desc',
          selectAllRows: 'Select all rows',
          selectRow: 'Select this row'
        },
        fixedLayout: true,
        items: this.state.items,
        id: "DataTableExample-2",
        onRowChange: this.handleChanged,
        selection: this.state.selection,
        selectRows: "checkbox",
        onRenderSubHeadingRow: CustomHeaderRow
      }, /*#__PURE__*/React.createElement(DataTableColumn, {
        label: "Name",
        primaryColumn: true,
        property: "opportunityName",
        width: "10rem"
      }), /*#__PURE__*/React.createElement(DataTableColumn, {
        label: "Account Name",
        property: "accountName",
        width: "8rem"
      }), /*#__PURE__*/React.createElement(DataTableColumn, {
        label: "Close Date",
        property: "closeDate"
      }), /*#__PURE__*/React.createElement(DataTableColumn, {
        label: "Stage",
        property: "stage"
      }), /*#__PURE__*/React.createElement(DataTableColumn, {
        label: "Confidence",
        property: "confidence"
      }), /*#__PURE__*/React.createElement(DataTableColumn, {
        label: "Amount",
        property: "amount"
      }), /*#__PURE__*/React.createElement(DataTableColumn, {
        label: "Contact",
        property: "contact"
      }))));
    }
  }]);

  return Example;
}(React.Component);

_defineProperty(Example, "displayName", 'DataTableExample');

export default Example; // export is replaced with `ReactDOM.render(<Example />, mountNode);` at runtime
