// @salesforce/design-system-react/module/components/data-table/__examples__/interactive-elements.js
function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function"); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, writable: true, configurable: true } }); if (superClass) _setPrototypeOf(subClass, superClass); }

function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf || function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }

function _createSuper(Derived) { var hasNativeReflectConstruct = _isNativeReflectConstruct(); return function _createSuperInternal() { var Super = _getPrototypeOf(Derived), result; if (hasNativeReflectConstruct) { var NewTarget = _getPrototypeOf(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return _possibleConstructorReturn(this, result); }; }

function _possibleConstructorReturn(self, call) { if (call && (_typeof(call) === "object" || typeof call === "function")) { return call; } return _assertThisInitialized(self); }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function () {})); return true; } catch (e) { return false; } }

function _getPrototypeOf(o) { _getPrototypeOf = Object.setPrototypeOf ? Object.getPrototypeOf : function _getPrototypeOf(o) { return o.__proto__ || Object.getPrototypeOf(o); }; return _getPrototypeOf(o); }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import React from 'react';
import { action } from '@storybook/addon-actions';
import DataTable from "../../../../components/data-table"; // `~` is replaced with design-system-react at runtime

import DataTableColumn from "../../../../components/data-table/column";
import DataTableCell from "../../../../components/data-table/cell";
import DataTableInteractiveElement from "../../../../components/data-table/interactive-element";
import DataTableInteractiveLink from "../../../../components/data-table/interactive-link";
import IconSettings from "../../../../components/icon-settings";
import Button from "../../../../components/button";
import Checkbox from "../../../../components/checkbox";
var InteractiveButton = DataTableInteractiveElement(Button);
var InteractiveCheckBox = DataTableInteractiveElement(Checkbox);

var CustomDataTableCell = function CustomDataTableCell(_ref) {
  var id = _ref.id,
      children = _ref.children,
      props = _objectWithoutProperties(_ref, ["id", "children"]);

  var cell = /*#__PURE__*/React.createElement(DataTableCell, props, /*#__PURE__*/React.createElement(InteractiveCheckBox, {
    id: "".concat(id, "-checkbox"),
    labels: {
      label: 'Option'
    }
  }), /*#__PURE__*/React.createElement(InteractiveButton, {
    onClick: action('button clicked'),
    label: "Open"
  }), /*#__PURE__*/React.createElement("div", null, /*#__PURE__*/React.createElement(DataTableInteractiveLink, {
    onClick: action('link clicked')
  }, "Click me")));
  return cell;
};

CustomDataTableCell.displayName = DataTableCell.displayName;
var columns = [/*#__PURE__*/React.createElement(DataTableColumn, {
  key: "opportunity",
  label: "Opportunity Name",
  property: "opportunityName",
  primaryColumn: true
}, /*#__PURE__*/React.createElement(CustomDataTableCell, null)), /*#__PURE__*/React.createElement(DataTableColumn, {
  key: "account-name",
  label: "Account Name",
  property: "accountName"
}), /*#__PURE__*/React.createElement(DataTableColumn, {
  key: "amount",
  label: "Amount",
  property: "amount"
}), /*#__PURE__*/React.createElement(DataTableColumn, {
  key: "contact",
  label: "Contact",
  property: "contact"
}, /*#__PURE__*/React.createElement(CustomDataTableCell, null))];

var Example = /*#__PURE__*/function (_React$Component) {
  _inherits(Example, _React$Component);

  var _super = _createSuper(Example);

  function Example() {
    var _this;

    _classCallCheck(this, Example);

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = _super.call.apply(_super, [this].concat(args));

    _defineProperty(_assertThisInitialized(_this), "state", {
      items: [{
        id: '8IKZHZZV80',
        opportunityName: 'Cloudhub',
        accountName: 'Cloudhub',
        closeDate: '4/14/2015',
        stage: 'Prospecting',
        confidence: '20%',
        amount: '$25k',
        contact: 'jrogers@cloudhub.com'
      }, {
        id: '5GJOOOPWU7',
        opportunityName: 'Cloudhub + Anypoint Connectors',
        accountName: 'Cloudhub',
        closeDate: '4/14/2015',
        stage: 'Prospecting',
        confidence: '20%',
        amount: '$25k',
        contact: 'jrogers@cloudhub.com'
      }, {
        id: '8IKZHZZV81',
        opportunityName: 'Cloudhub',
        accountName: 'Cloudhub',
        closeDate: '4/14/2015',
        stage: 'Prospecting',
        confidence: '20%',
        amount: '$25k',
        contact: 'jrogers@cloudhub.com'
      }]
    });

    return _this;
  }

  _createClass(Example, [{
    key: "render",
    value: function render() {
      return /*#__PURE__*/React.createElement(IconSettings, {
        iconPath: "/assets/icons"
      }, /*#__PURE__*/React.createElement("div", {
        style: {
          overflow: 'auto'
        }
      }, /*#__PURE__*/React.createElement("h3", {
        className: "slds-text-heading_medium slds-m-vertical_medium"
      }, "Default Fluid Layout"), /*#__PURE__*/React.createElement(DataTable, {
        items: this.state.items,
        id: "DataTableExample-1-default",
        fixedLayout: true,
        keyboardNavigation: true
      }, columns), /*#__PURE__*/React.createElement("h3", {
        className: "slds-text-heading_medium slds-m-vertical_medium"
      }, "Striped"), /*#__PURE__*/React.createElement(DataTable, {
        items: this.state.items,
        id: "DataTableExample-1-striped",
        striped: true,
        fixedLayout: true,
        keyboardNavigation: true
      }, columns), /*#__PURE__*/React.createElement("h3", {
        className: "slds-text-heading_medium slds-m-vertical_medium"
      }, "No Row Hover"), /*#__PURE__*/React.createElement(DataTable, {
        items: this.state.items,
        id: "DataTableExample-noRowHover",
        noRowHover: true,
        fixedLayout: true,
        keyboardNavigation: true
      }, columns), /*#__PURE__*/React.createElement("h3", {
        className: "slds-text-heading_medium slds-m-vertical_medium"
      }, "Column Bordered"), /*#__PURE__*/React.createElement(DataTable, {
        columnBordered: true,
        items: this.state.items,
        id: "DataTableExample-columnBordered",
        fixedLayout: true,
        keyboardNavigation: true
      }, columns), /*#__PURE__*/React.createElement("h3", {
        className: "slds-text-heading_medium slds-m-vertical_medium"
      }, "Resizable columns"), /*#__PURE__*/React.createElement(DataTable, {
        items: this.state.items,
        id: "DataTableExample-column-resizable",
        fixedLayout: true,
        fixedHeader: true,
        resizable: true,
        keyboardNavigation: true
      }, columns)));
    }
  }]);

  return Example;
}(React.Component);

_defineProperty(Example, "displayName", 'DataTableExample');

export default Example; // export is replaced with `ReactDOM.render(<Example />, mountNode);` at runtime
